### PR TITLE
Convert site store methods to coroutines

### PR DESCRIPTION
**Pull request by planarvoid**
This PR converts the first methods to fetch site, fetch sites, create new site and fetch site formats to coroutines. I've changed both the WPCom and the XMLRPC rest clients whenever possible. This PR could be tested with the related WPAndroid changes - https://github.com/wordpress-mobile/WordPress-Android/pull/14812

### CODE REVIEW

File: fluxc/src/main/java/org/wordpress/android/fluxc/network/rest/wpcom/WPComGsonRequestBuilder.kt [hunk 1]
```diff
@@ -91,13 +92,17 @@ class WPComGsonRequestBuilder
         url: String,
         params: Map<String, String>?,
         body: Map<String, Any>?,
-        clazz: Class<T>
+        clazz: Class<T>,
+        retryPolicy: RetryPolicy? = null
```
Comment (develric) [inline fluxc/src/main/java/org/wordpress/android/fluxc/network/rest/wpcom/WPComGsonRequestBuilder.kt:97]: maybe we could add a comment in the java doc about the rationale of this param, wdyt?

File: fluxc/src/main/java/org/wordpress/android/fluxc/store/SiteStore.kt [hunk 2]
```diff
@@ -1133,11 +1146,15 @@ class SiteStore
         when (actionType) {
             FETCH_PROFILE_XML_RPC -> fetchProfileXmlRpc(action.payload as SiteModel)
             FETCHED_PROFILE_XML_RPC -> updateSiteProfile(action.payload as SiteModel)
-            FETCH_SITE -> fetchSite(action.payload as SiteModel)
-            FETCH_SITES -> fetchSites(action.payload as FetchSitesPayload)
-            FETCHED_SITES -> handleFetchedSitesWPComRest(action.payload as SitesModel)
-            FETCH_SITES_XML_RPC -> fetchSitesXmlRpc(action.payload as RefreshSitesXMLRPCPayload)
-            FETCHED_SITES_XML_RPC -> updateSites(action.payload as SitesModel)
+            FETCH_SITE -> coroutineEngine.launch(T.MAIN, this, "Fetch site") {
+                fetchSite(action.payload as SiteModel)
```
Comment (develric) [inline fluxc/src/main/java/org/wordpress/android/fluxc/store/SiteStore.kt]: could it make sense to move the `emitChange()` call here as well?

File: fluxc/src/main/java/org/wordpress/android/fluxc/network/rest/wpcom/site/SiteRestClient.kt [hunk 3]
```diff
@@ -116,38 +118,31 @@ class SiteRestClient @Inject constructor(
     data class IsWPComResponsePayload(
         val url: String,
         val isWPCom: Boolean = false
-    ) : Payload<BaseNetworkError>()
+    ) : Payload<BaseNetworkError?>()
```
Comment (develric) [inline fluxc/src/main/java/org/wordpress/android/fluxc/network/rest/wpcom/site/SiteRestClient.kt]: Possibly it's just obvious 😄 , but why do we need it nullable here?

**Review by develric — COMMENTED** [review]
Great job @planarvoid , going to check the WPAndroid companion PR, but wanted to left just a couple of comments to double check with you 🙇

File: fluxc/src/main/java/org/wordpress/android/fluxc/network/rest/wpcom/site/SiteRestClient.kt (hunk 3 above)
Comment (planarvoid) [reply fluxc/src/main/java/org/wordpress/android/fluxc/network/rest/wpcom/site/SiteRestClient.kt]: we don't, good catch. I think it's an overlook that was caused by splitting the PR into several small ones and merging between them :-). 
Fixed in 5de8dde

File: fluxc/src/main/java/org/wordpress/android/fluxc/store/SiteStore.kt (hunk 2 above)
Comment (planarvoid) [reply fluxc/src/main/java/org/wordpress/android/fluxc/store/SiteStore.kt]: makes sense 👍
Changed in 1a48168

File: fluxc/src/main/java/org/wordpress/android/fluxc/network/rest/wpcom/WPComGsonRequestBuilder.kt (hunk 1 above)
Comment (planarvoid) [reply fluxc/src/main/java/org/wordpress/android/fluxc/network/rest/wpcom/WPComGsonRequestBuilder.kt:97]: Added in 9876e32

**Comment by planarvoid** [comment]
Thanks for the review @develric , I've implemented all your suggestions 👍

**Review by develric — APPROVED** [review]
Thanks for the changes 🙇 ! Approving this one 👍 !

**Pull request merged**